docs/modules/ROOT/pages/api/erc1155.adoc

```diff
@@ -0,0 +1,664 @@
+* xref:#ERC1155Component-balance_of_batch[`++balance_of_batch(self, accounts, token_ids)++`]
+* xref:#ERC1155Component-safe_transfer_from[`++safe_transfer_from(self, from, to, token_id, value, data)++`]
+* xref:#ERC1155Component-safe_batch_transfer_from[`++safe_batch_transfer_from(self, from, to, token_ids, values, data)++`]
+* xref:#ERC1155Component-is_approved_for_all[`++is_approved_for_all(self, owner, operator)++`]
+* xref:#ERC1155Component-set_approval_for_all[`++set_approval_for_all(self, operator, approved)++`]
+
+[.sub-index#ERC1155Component-Embeddable-Impls-ERC1155MetadataURIImpl]
+.ERC1155MetadataURIImpl
+* xref:#ERC1155Component-uri[`++uri(self, token_id)++`]
+--
+
+[.contract-index#ERC1155Component-Embeddable-Impls-camelCase]
+.Embeddable implementations (camelCase)
+--
+[.sub-index#ERC1155Component-Embeddable-Impls-ER1155CamelImpl]
+.ER1155CamelImpl
+* xref:#ERC1155Component-balanceOf[`++balanceOf(self, account, tokenId)++`]
+* xref:#ERC1155Component-balanceOfBatch[`++balanceOfBatch(self, accounts, tokenIds)++`]
+* xref:#ERC1155Component-safeTransferFrom[`++safeTransferFrom(self, from, to, tokenId, value, data)++`]
+* xref:#ERC1155Component-safeBatchTransferFrom[`++safeBatchTransferFrom(self, from, to, tokenIds, values, data)++`]
+* xref:#ERC1155Component-isApprovedForAll[`++isApprovedForAll(self, owner, operator)++`]
+* xref:#ERC1155Component-setApprovalForAll[`++setApprovalForAll(self, operator, approved)++`]
+--
+
+[.contract-index]
+.Internal Functions
+--
+.InternalImpl
+* xref:#ERC1155Component-initializer[`++initializer(self, uri)++`]
+* xref:#ERC1155Component-update[`++update(self, from, to, token_ids, values)++`]
+* xref:#ERC1155Component-update_with_acceptance_check[`++update_with_acceptance_check(self, from, to, token_ids, values, data)++`]
+* xref:#ERC1155Component-mint_with_acceptance_check[`++mint_with_acceptance_check(self, to, token_id, value, data)++`]
+* xref:#ERC1155Component-batch_mint_with_acceptance_check[`++batch_mint_with_acceptance_check(self, to, token_ids, values, data)++`]
+* xref:#ERC1155Component-burn[`++burn(self, from, token_id, value)++`]
+* xref:#ERC1155Component-batch_burn[`++batch_burn(self, from, token_ids, values)++`]
+* xref:#ERC1155Component-set_uri[`++set_uri(self, uri)++`]
+--
+
+[.contract-index]
+.Events
+--
+.IERC1155
+* xref:#ERC1155Component-TransferSingle[`++TransferSingle(operator, from, to, id, value)++`]
+* xref:#ERC1155Component-TransferBatch[`++TransferBatch(operator, from, to, ids, values)++`]
+* xref:#ERC1155Component-ApprovalForAll[`++ApprovalForAll(owner, operator, approved)++`]
+* xref:#ERC1155Component-URI[`++URI(value, id)++`]
+--
+
+==== Embeddable functions
+
+[.contract-item]
+[[ERC1155Component-balance_of]]
+==== `[.contract-item-name]#++balance_of++#++(self: @ContractState, account: ContractAddress, token_id: u256) → u256++` [.item-kind]#external#
+
+Returns the number of NFTs owned by `account` for a specific `token_id`.
```

Comment:
i think it's a bit odd to have different definitions for the same methods in here and in the interface. why not following ERC721's model where we point to the definition in the interface, and specify impl-specific behavior such as input requirements, emitted events, etc?

Reply:
Unnecessary extra click IMO, and more when the interface description is not the same as the implementation description (lacking requirements, etc...). Users needed to click to go to the interface to get the main idea, and then get back to the component to see the extra requirements and things, feels like not the best UX IMO. I vote for avoiding those redirections as much as possible, and I think this is a good example of how we can avoid it.

Reply:
For example, in ERC721 we have requirements in snake_case methods, but not in camelCase methods, because in camelCase we are redirecting to the interface description, not to the component snake_case description, so users checking camelCase would be totally missing the requirements.

Reply:
I think we should explain everything in the snake case section, not redirecting to the interface, and then in camelCase section we can redirect to the snake case section, but not to the interface.